### ISSUE TITLE
Unxpected highlighting for if statement within HTML tag

### ISSUE THREAD

**Issue by wjharney** [ventojs/vscode-vento#8]
For `<a {{ if foo == bar }}attribute{{ /if }}>`:
* `==` is two colors
* `bar` is not the same color as `foo`
* `/if` is not the same color as `/if`
Ideally, both parts of each pair should match.

Also notable, but less distracting, the colors used within an HTML tag do not match the colors used for regular content, particularly the content within the if block (`attribute`).

<img width="318" alt="screenshot of the described issue" src="https://github.com/user-attachments/assets/6e27eec6-6c96-427d-836b-01038754e8ff">

**Comment by oscarotero** [comment]
Yes, I'm aware of this, and it depends on the position of the js code in the HTML:
<img width="708" alt="image" src="https://github.com/user-attachments/assets/8629d0f5-a052-47f0-9597-3f62ecf645d1">

- If it's inside an attribute, treats the vento code as a string.
- If it's inside a tag, does what you are saying
- If it's outside a tag, it works fine

And as an extra annoying case, if it's inside an attribute and it has quotes, the highlighting interprets these quotes as the end of the attribute:
<img width="733" alt="image" src="https://github.com/user-attachments/assets/ca364d75-8574-4b93-9c60-b4a0e1451a4c">

Unfortunately, I'm not an expert creating VSCode plugins. If anyone want to help with this, I'd appreciate.

**Comment by bobrocke** [comment]
For the code:
```
<p class="{{ foo == bar ? foo : bar }}">This is a paragraph</p>
<p {{ foo == bar ? foo : bar }}>This is a paragraph</p>
<p> {{ foo == bar ? foo : bar }}This is a paragraph</p>
<p class="{{ foo == bar ? "foo" : "bar" }}">This is a paragraph</p>
```

I see this with version 0.2.3 of the Vento language:

<img width="501" height="86" alt="Image" src="https://github.com/user-attachments/assets/401ff635-c724-4d9f-a3fe-49dbd5fa7361" />

So only the second instance looks 'wrong'. Is that how you see it?

**Comment by oscarotero** [comment]
@bobrocke Yeah, I was able to fix the case of vento code inside html attributes some time ago (commit https://github.com/ventojs/vscode-vento/commit/a310336f67890805ade8303cbbf9d721338e9388).
But couldn't figure out how to priorize Vento syntax inside an HTML tag (which I suspect is the cause of the second instance issue).

**Comment by bobrocke** [comment]
After several rounds with Claude Opus 4.5, I've managed to get to:

<img width="510" height="136" alt="Image" src="https://github.com/user-attachments/assets/3a617869-4987-4807-a16e-63f68c28eaba" />

for:
```
<a {{ if foo == bar }}attribute{{ /if }}></a>
{{ if foo == bar }}attribute{{ /if }}

<p class="{{ foo == bar ? foo : bar }}">This is a paragraph</p>
<p {{ foo == bar ? foo : bar }}>This is a paragraph</p>
<p> {{ foo == bar ? foo : bar }}This is a paragraph</p>
<p class="{{ foo == bar ? "foo" : "bar" }}">This is a paragraph</p>
```

Claude suggests that VS Code's TextMate grammars can't do everything Sublime Text grammars can, and that the Sublime Text Vento language can handle all the cases above.

Would you like a PR for what I've been able to accomplish, or wait for a more complete solution, if there is one?

**Comment by RedCMD** [comment]
its broken cause its only being highlighted with html

<img width="1322" height="212" alt="Image" src="https://github.com/user-attachments/assets/3746a4f8-9d37-4b10-b957-631db8bb1e72" />

https://github.com/ventojs/vscode-vento/blob/05e946486c11d91244d443075286f9151b0ac4e2/syntaxes/vento.tmLanguage.json#L11

<img width="576" height="284" alt="Image" src="https://github.com/user-attachments/assets/4f578e65-c90e-47ac-b426-1a4d4a1169e0" />

https://github.com/ventojs/vscode-vento/blob/05e946486c11d91244d443075286f9151b0ac4e2/package.json#L36-L42

fixing the `"injectTo"` will make it work in language `html`
(and `"embeddedLanguages"` should target the `languageId`, not `scopeName`)

<img width="954" height="218" alt="Image" src="https://github.com/user-attachments/assets/3362fedc-c6b2-4f47-819e-c144ec0df462" />

however this rule in html gobbles up that 2nd occurrence

<img width="1659" height="488" alt="Image" src="https://github.com/user-attachments/assets/33f6de31-51d2-4e60-97e8-e496202565d6" />

---

to fix it in language `vento`
you will need to add the different `meta.tags` to `"injections"`
keep in mind VSCode does not support the any scope `*`
`string.quoted.*.html` is treated as two different consecutive scopes `string.quoted.` and `.html`

https://github.com/ventojs/vscode-vento/blob/05e946486c11d91244d443075286f9151b0ac4e2/syntaxes/vento.tmLanguage.json#L4-L10

<img width="1157" height="246" alt="Image" src="https://github.com/user-attachments/assets/3663b3df-3415-4d76-a742-6ef71e448fa1" />

```json
"L:source.vento (meta.tag.metadata | meta.tag.structure | meta.tag.inline | meta.tag.object | meta.tag.other | meta.tag.custom) -meta.embedded -string -comment": {
	"patterns": [ { "include": "source.vento" } ]
}
```

<img width="958" height="219" alt="Image" src="https://github.com/user-attachments/assets/fc247454-5912-427f-8de2-6f6e38dfbec8" />

---

if you want correct `vento` `language.confiuration.json` settings within html
brackets etc
you'll have to ask them to add you to their `"embeddedLanguages"`
and add the scope to your grammar for them to target 
<img width="707" height="416" alt="Image" src="https://github.com/user-attachments/assets/a9cb50dc-51d4-41b4-bb5c-3ceb32d93ad6" />

---

[vento-html.tmLanguage.json](https://github.com/ventojs/vscode-vento/blob/main/syntaxes/vento-html.tmLanguage.json) never gets used?

**Comment by bobrocke** [comment]
Maybe so. Would you take a stab at a PR?
